gateway-ha/src/main/java/io/trino/gateway/ha/security/ResourceSecurityDynamicFeature.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.security;
+
+import jakarta.annotation.Priority;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.container.DynamicFeature;
+import jakarta.ws.rs.container.ResourceInfo;
+import jakarta.ws.rs.core.FeatureContext;
+
+import static jakarta.ws.rs.Priorities.AUTHENTICATION;
+import static java.util.Objects.requireNonNull;
+
+@Priority(AUTHENTICATION)
+public class ResourceSecurityDynamicFeature
```

Comment:
I am hoping you can think of a better class name here .. but maybe its just my unfamiliarity with the jax-rs api

Reply:
https://github.com/trinodb/trino/blob/569b045f3bb228703f93a4784411274df5f597d7/core/trino-main/src/main/java/io/trino/server/security/ResourceSecurityDynamicFeature.java
I use the same class name as Trino.

Reply:
ok .. that works then.